### PR TITLE
Implement dot-div (.÷) operator, corresponding to ./ or .%

### PR DESCRIPTION
**Pull request by eschnett**

### CODE REVIEW

File: base/functors.jl [hunk 1]
```diff
@@ -67,6 +67,9 @@ call(::LDivFun, x, y) = x \ y
 immutable IDivFun <: Func{2} end
 call(::IDivFun, x, y) = div(x, y)
 
+immutable DotIDivFun <: Func{2} end
+call(::IDivFun, x, y) = x .÷ y
```
Comment (tkelman) [inline base/functors.jl]: `DotIDivFun`

Comment (eschnett) [inline base/functors.jl]: I followed `DotRemFun`'s implementation. Should that also be changed to `DotRemFun`?

Comment (tkelman) [inline base/functors.jl]: Yeah I suspect that's a typo? And a sign that the tests of broadcasting `rem` aren't thorough enough to catch the difference between `rem(x,y)` and `x.%y`.

**Comment by stevengj** [comment]
Seems reasonable.  In the long run, however, I'd prefer to update the parser so that for any binary operator `⧆` it automatically also parses a `.⧆` operator with the same precedence.

**Comment by StefanKarpinski** [comment]
It's arguable that `==` and `.==`, `<` and `.<` etc. shouldn't have the same precedence.

**Pull request merged**